src/shared/find-esp.c

```diff
@@ -767,6 +772,15 @@ static int verify_xbootldr(
         if (relax_checks)
                 goto finish;
 
+        if (devnum_is_zero(devid))
+                return log_full_errno(searching ? LOG_DEBUG : LOG_ERR,
+                                      SYNTHETIC_ERRNO(searching ? EADDRNOTAVAIL : ENODEV),
+                                      "Could not determine backing block device of directory \"%s\" (btrfs RAID?).%s",
+                                      p,
+                                      searching ? "" :
+                                      "Hint: set $SYSTEMD_RELAX_XBOOTLDR_CHECKS=yes environment variable "
```

Comment:
Please prefix with "\n" here, otherwise the two strings run together.

Reply:
Oh, I mistakenly dropped `\n`.